File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/repository/WordsRepository.java
package com.przemyslawlusnia.vocabularycreator.wordlist.repository;

import com.przemyslawlusnia.vocabularycreator.wordlist.domain.WordDomainModel;
import java.util.List;
import rx.Observable;

public interface WordsRepository {
  Observable<Boolean> delete(List<WordDomainModel> words);

  Observable<Boolean> add(WordDomainModel wordDomainModel);

  Observable<List<WordDomainModel>> getAllWords();
}

File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/repository/WordsRepository.kt
package com.przemyslawlusnia.vocabularycreator.wordlist.repository

import com.przemyslawlusnia.vocabularycreator.wordlist.domain.WordDomainModel
import rx.Observable

interface WordsRepository {
    fun delete(words: List<WordDomainModel>): Observable<Void>

    fun add(wordDomainModel: WordDomainModel): Observable<Boolean>

    val allWords: Observable<List<WordDomainModel>>
}
